AwsCryptographicMaterialProviders/dafny/AwsCryptographyKeyStoreAdmin/src/InitializeMutation.dfy

```diff
@@ -0,0 +1,612 @@
+    requires
+      && input.storage.ValidState()
+      && match input.keyManagerStrategy {
+           case reEncrypt(km) => km.kmsClient.ValidState() && AwsKmsUtils.GetValidGrantTokens(Some(km.grantTokens)).Success?
+           case decryptEncrypt(kmD, kmE) =>
+             && kmD.kmsClient.ValidState() && kmE.kmsClient.ValidState()
+             && AwsKmsUtils.GetValidGrantTokens(Some(kmD.grantTokens)).Success?
+             && AwsKmsUtils.GetValidGrantTokens(Some(kmE.grantTokens)).Success?
+         }
+      && input.SystemKey.ValidState()
+      && input.ValidState()
+    ensures
+      && input.storage.ValidState()
+      && input.SystemKey.ValidState()
+      &&
+         match input.keyManagerStrategy {
+           case reEncrypt(km) => km.kmsClient.ValidState()
+           case decryptEncrypt(kmD, kmE) => kmD.kmsClient.ValidState() && kmE.kmsClient.ValidState()
+         }
+      && input.ValidState()
+    modifies
+      input.storage.Modifies,
+             match input.keyManagerStrategy {
+               case reEncrypt(km) => km.kmsClient.Modifies
+               case decryptEncrypt(kmD, kmE) => kmD.kmsClient.Modifies + kmE.kmsClient.Modifies
+             },
+             input.SystemKey.Modifies
+  {
+    var resumeMutation? := false;
+
+    // Fetch Active Branch Key & Beacon Key & Mutation Lock
+    var readItems? := input.storage.GetItemsForInitializeMutation(
+      Types.AwsCryptographyKeyStoreTypes.GetItemsForInitializeMutationInput(Identifier := input.Identifier));
+    var readItems :- readItems?
+    .MapFailure(e => Types.Error.AwsCryptographyKeyStore(e));
+
+    if (readItems.MutationCommitment.None? && readItems.MutationIndex.Some?) {
+      var indexUUID := readItems.MutationIndex.value.UUID;
+      return Failure(
+          Types.MutationInvalidException(
+            message := "Found a Mutation Index but no Mutation Commitment."
+            + " The Key Store's Storage, for this Branch Key, has become corrupted."
+            + " Recommend auditing the Branch Key's items for tampering."
+            + " Recommend auditing access to the storage."
+            + " To successfully start a new mutation, delete the Mutation Index."
+            + " But know that the new mutation will fail if any corrupt items are encountered."
+            + "\nBranch Key ID: " + input.Identifier + ";"
+            + " Mutation Index UUID: " + indexUUID));
+    }
+
+    if (readItems.MutationCommitment.Some?) {
+      resumeMutation? :- CommitmentAndInputMatch(
+        internalInput := input,
+        commitment := readItems.MutationCommitment.value);
+      if (resumeMutation?) {
+        // TODO-SystemKey :: ResumeMutation will validate SystemKey
+        output := ResumeMutation(
+          commitment := readItems.MutationCommitment.value,
+          index := readItems.MutationIndex,
+          logicalKeyStoreName := input.logicalKeyStoreName,
+          storage := input.storage, // );//,
+          SystemKey := input.SystemKey);
```

Comment:
```suggestion
          storage := input.storage,
          systemKey := input.SystemKey);
```

Reply:
This change would require changing the method/function's signature as well.

Reply:
it doesn't look like a finished product if we mix casing here and there...

Reply:
I will refactor the signature.